### DISCUSSION TITLE
Set Default Values for Survey Variables in Scheduled Tasks

### DISCUSSION THREAD

**Discussion by lmrlyv** [semaphoreui/semaphore#2226] [Q&A]
I use **Survey Variables** in the Task Template to set values for the variables defined in the _vars_prompt_ section of the playbook. This works fine when I manually trigger the task and enter the values by hand. However, I haven't found a way to set these variables when the task template is scheduled to run automatically. Is there currently a way to achieve this? If not, wouldn't it be better to allow defining default values when creating a schedule?

**Comment by lmrlyv** [comment]
Hi, @fiftin, @matejkramny, and @strangeman. Sorry to push for a response, but I'd really appreciate your thoughts on this question/idea. What do you think?

**Comment by northwestnodes-eric** [comment]
A bit of a late reply, but no one here seems interested in really helping each other. You can set default variables for the playbook itself like so: `hosts: "{{ target_hosts | default('servers') }}"` where you can set a `target_hosts` survey variable as needed and leave it empty to default to 'servers'. Might not work for your situation, but I see no other option at the moment.

**Reply by lmrlyv** [reply]
Thank you for the reply though. 

Yeah, it feels like the project is abandoned or has very little development going on.

I don’t remember exactly but if I am not mistaken, the playbook execution gets stuck on the user input indefinitely when it is triggered on schedule even though the default value for vars_prompt is defined in the playbook itself.

I will try this out again tomorrow and update here the exact behavior

**Reply by jengstrom440** [reply]
Hello, wondering if you had any success? I am looking at several different scenarios and I really wish the surveys worked like Ansible Automation Platform 2.4 or AWX 18+. I would also like to create a default value for a survey question, but should I wish upon launch, change it to something else; otherwise use the answer I provided. I don't want to put it in the playbook though because I need several different playbooks then!

**Reply by mg-oerp** [reply]
I'm running into similar issues but with trying to figure out how to pass static variables to a schedule.   I don't want to end up having to duplicate the same playbook with hard coded bits a hundred times just because I want to schedule them...

> Thank you for the reply though.
> 
> Yeah, it feels like the project is abandoned or has very little development going on.
> 
> I don’t remember exactly but if I am not mistaken, the playbook execution gets stuck on the user input indefinitely when it is triggered on schedule even though the default value for vars_prompt is defined in the playbook itself.
> 
> I will try this out again tomorrow and update here the exact behavior

I'm starting to think the same thing.  It seems like things have been added semi-recently, but there also seems to be some large gaps in what it can and can't do and it doesn't seem like it's got much momentum anymore.  Thinking maybe this is not the solution I was hoping it might become.

**Reply by robinmalik** [reply]
With 67 open pull requests, it also makes it off-putting to consider putting in the work to raise one yourself.

**Reply by mg-oerp** [reply]
Yes exactly my point.  I found (and fixed) a few issues myself and haven't bothered to submit them here sadly.  Ultimately am moving along to a different solution because the devs seem to have their own agenda and could care less about anything people contribute either idea OR code wise.  It seems like they are just in it to try and cash in on the month fees for paying customers? 

Oh yes and don't mention the 800+ issues that are mostly stale and ignored.

**Reply by sebvdw** [reply]
Does this kind of thing happen when a project doesnt have enough maintainers? 

I was interested in contributing but now I am re-considering...

also facing this problem: wanting to pass static variables to scheduled tasks.

**Reply by Gerk** [reply]
I get the feeling that these developers don't really want other people helping them given the number of PRs and issues that they have just plain ignored.  I went onto several places where they offer support and one of the devs told me that this (github) is the most active place to get support and/or help and/or to contribute.  So I came here and got ignored just like the other places LOL.

They are trying to do a commercial venture with this (and the terms of it don't make sense to me but maybe it does for a specific demographic), so maybe they are being extra guarded around the code and only want to add the things that they want to add, which is fine, BUT ... ignoring the community is a pretty terrible way of doing this.  I submitted a couple of PRs on another account that got ignored so I ended up just pulling them and moving along.

That's the thing about "open source" -- while most devs are pretty happy to have help (or at least gracious about it) not everyone is.  I think these guys are brothers (or at least a couple of them are) and they seem to want to "keep it in the family", which is also fine, but they should at least be up front and say that they don't want PRs instead of letting people go through all the effort and then just ignoring it until it goes stale.
